nix/modules/flake-parts/lib.nix

```diff
@@ -22,6 +22,25 @@ let
         }
       ];
     };
+
+    # Common and useful setting across all platforms
+    common = { flake, lib, ... }: {
+      nix = {
+        # Enables use of `nix-shell -p ...` etc
+        nixPath = [ "nixpkgs=${flake.inputs.nixpkgs}" ];
+        # Make `nix shell` etc use pinned nixpkgs
+        registry.nixpkgs.flake = flake.inputs.nixpkgs;
```

Comment:
I guess we don't need this anymore:

- NixOS: https://github.com/NixOS/nixpkgs/pull/254405
- nix-darwin: https://github.com/LnL7/nix-darwin/pull/1067